packages/ssz/src/type/listComposite.ts

```diff
@@ -0,0 +1,172 @@
+{
+  readonly typeName: string;
+  readonly itemsPerChunk = 1;
+  readonly depth: number;
+  readonly chunkDepth: number;
+  readonly maxChunkCount: number;
+  readonly fixedSize = null;
+  readonly minSize = 0;
+  readonly maxSize: number;
+
+  constructor(readonly elementType: ElementType, readonly limit: number) {
+    super();
+
+    if (elementType.isBasic) {
+      throw Error("elementType must not be basic");
+    }
+
+    this.typeName = `List[${elementType.typeName}, ${limit}]`;
+    this.maxChunkCount = this.limit;
+    this.chunkDepth = maxChunksToDepth(this.maxChunkCount);
+    // Depth includes the extra level for the length node
+    this.depth = this.chunkDepth + 1;
+    this.maxSize = this.limit * elementType.maxSize;
+  }
+
+  get defaultValue(): ValueOf<ElementType>[] {
+    return [];
+  }
+
+  getView(tree: Tree): ListCompositeTreeView<ElementType> {
+    return new ListCompositeTreeView(this, tree);
+  }
+
+  getViewDU(node: Node, cache?: unknown): ListCompositeTreeViewDU<ElementType> {
+    // cache type should be validated (if applicate) in the view
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return new ListCompositeTreeViewDU(this, node, cache as any);
+  }
+
+  commitView(view: ListCompositeTreeView<ElementType>): Node {
+    return view.node;
+  }
+
+  commitViewDU(view: ListCompositeTreeViewDU<ElementType>): Node {
+    return view.commit();
+  }
+
+  cacheOfViewDU(view: ListCompositeTreeViewDU<ElementType>): unknown {
+    return view.cache;
+  }
+
+  // Serialization + deserialization
+
+  value_serializedSize(value: ValueOf<ElementType>[]): number {
+    return value_serializedSizeArrayComposite(this.elementType, value.length, value);
+  }
+
+  value_serializeToBytes(output: ByteViews, offset: number, value: ValueOf<ElementType>[]): number {
+    return value_serializeToBytesArrayComposite(this.elementType, value.length, output, offset, value);
+  }
+
+  value_deserializeFromBytes(data: ByteViews, start: number, end: number): ValueOf<ElementType>[] {
+    return value_deserializeFromBytesArrayComposite(this.elementType, data, start, end, this);
+  }
+
+  tree_serializedSize(node: Node): number {
+    const chunksNode = this.tree_getChunksNode(node);
+    const length = this.tree_getLength(node);
+    return tree_serializedSizeArrayComposite(this.elementType, length, this.chunkDepth, chunksNode);
+  }
+
+  tree_serializeToBytes(output: ByteViews, offset: number, node: Node): number {
+    const chunksNode = this.tree_getChunksNode(node);
+    const length = this.tree_getLength(node);
+    return tree_serializeToBytesArrayComposite(this.elementType, length, this.chunkDepth, chunksNode, output, offset);
+  }
+
+  tree_deserializeFromBytes(data: ByteViews, start: number, end: number): Node {
+    return tree_deserializeFromBytesArrayComposite(this.elementType, this.chunkDepth, data, start, end, this);
+  }
+
+  // Helpers for TreeView
+
+  tree_getLength(node: Node): number {
```

Comment:
is this method needed? same with listComposite

Reply:
Yes it's used by VIews